python/cudf/cudf/tests/test_string.py

```diff
@@ -852,6 +852,23 @@ def test_string_contains(ps_gs, pat, regex, flags, flags_raise, na, na_raise):
         assert_eq(expect, got)
 
 
+@pytest.mark.parametrize(
+    "data", [["hello", "world", None, "", "!"]],
+)
+@pytest.mark.parametrize(
+    "repeats",
+    [2, 0, -3, [5, 4, 3, 2, 6], [0, 0, 0, 0, 0], [-1, -2, -3, -4, -5]],
```

Comment:
What happen if one of the repeat values is None?

Reply:
```
import cudf
gs = cudf.Series(["hello", None, "bye"])
gs.str.repeat(5)
0    hellohellohellohellohello
1                         <NA>
2              byebyebyebyebye
dtype: object
```
versus
```
import pandas as pd
ps = pd.Series(["hello", None, "bye"])
0    hellohellohellohellohello
1                         None
2              byebyebyebyebye
dtype: object
```
so they're not exactly equal, but `cudf.str.repeat` can still handle None.

Reply:
Great, thanks! This is the expected behavior 👍 

It would be nice to include a None in one of the tests. Otherwise, this LGTM.

Reply:
oops, I just realized that the example I gave you was when one of the strings is None, and not when one of the repeat numbers is None. but it still exhibits the expected behavior: 
```
import cudf
s = cudf.Series(['hello', 'world'])
s.str.repeat([4, None])
0    hellohellohellohello
1                    <NA>
dtype: object
```
and
```
import pandas as pd
p = pd.Series(['hello', 'world'])
p.str.repeat([4, None])
0    hellohellohellohello
1                    None
dtype: object
```
and ok, I'll update the test with that!

Reply:
updated - I also realized that even though something like `repeats=[2,None]` works, `repeats=None` fails because it expects a number. I made the appropriate changes in `string.py` but lmk if you think there's a better way to handle this case.

Reply:
What is the expected behavior of `repeats=None`?

Reply:
in pandas you get: 
```
import pandas as pd
p = pd.Series(['hello', 'world'])
p.str.repeat(None)
0   NaN
1   NaN
dtype: float64
```
in order to pass the test I also had to add `return result.astype("float64") if none_flag else result` since the cuDF version would return `dtype: object` instead.

Reply:
Thanks. We actually want an "object" (string) column as the result here. Pandas doesn't have [nullable string types](https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html), but we do. We might want to return all NAs here (loosely the equivalent) or throw an error. @brandon-b-miller what do you think?

Reply:
also, for some reason if you use a list of None's in pandas, you get: 
```
import pandas as pd
p = pd.Series(['hello', 'world'])
p.str.repeat([None, None])
0    None
1    None
dtype: object
```

Reply:
I usually appeal to the pandas nullable string dtype for guidance in these cases. Since it's still experimental it sometimes has bugs that we shouldn't replicate. But here I think it helps:

```
x = pd.Series(['hello', 'dave'], dtype=pd.StringDtype())
x.str.repeat(None)
# TypeError: 'NoneType' object cannot be interpreted as an integer
```

Seems it tries to `int` the arg here as a guard.

Reply:
ok, so should I also allow `cudf.str.repeat` to error in this case then?

Reply:
I would be ok with that in this case, the main reason being that I can't think of what the expected behavior would really even be. One can maybe argue that nulls beget nulls and the result should be `<NA>` but historically our approach for this kind of edge case has been to throw and address it if someone comes looking for it specifically.

Reply:
ok, updated. thank you!